Rename things with more friendly names

hug-dev opened the pull request:
Fix #26 
It is based on PR #32 
Also start some work for #32 and #27 by creating the template files. They are obviously wrong now but can be changed when tackling those issues.

(just focus on the second commit for the review)

hug-dev commented on src/subcommands/delete_key.rs:
```diff
@@ -16,27 +16,25 @@ use structopt::StructOpt;
 /// Destroys a key.
 #[derive(Debug, StructOpt)]
 #[structopt(name = "destroy-key")]
```
ah I need also to remove those lines everywhere... they are not used

ionut-arm commented on src/subcommands/create_ecc_key.rs:
```diff
@@ -0,0 +1,90 @@
+//! on how it will be possible in the future to generate a key from the command line. This is
+//! currently useful for playing with the tool, and demonstrating the use of Parsec.
+//!
+//! This will generate a 2048 bits RSA key pair for signing.
```
Will it, though?

hug-dev replied:
Yeah the text and content of all of the new files is wrong, planning to change it in a subsequent PR

ionut-arm commented on src/subcommands/create_ecc_key.rs:
```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Contributors to the Parsec project.
```
```suggestion
// Copyright 2021 Contributors to the Parsec project.
```

ionut-arm commented on src/subcommands/create_ecc_key.rs:
```diff
@@ -0,0 +1,90 @@
+                key_name: psa_generate_key_subcommand.key_name.clone(),
+                attributes: Attributes {
+                    lifetime: Lifetime::Persistent,
+                    key_type: Type::RsaKeyPair,
+                    bits: 2048,
+                    policy: Policy {
+                        usage_flags: UsageFlags {
+                            sign_hash: true,
+                            ..Default::default()
+                        },
+                        permitted_algorithms: AsymmetricSignature::RsaPkcs1v15Sign {
+                            hash_alg: Hash::Sha256.into(),
+                        }
+                        .into(),
```
Copypasta

ionut-arm commented on src/subcommands/create_ecc_key.rs:
```diff
@@ -0,0 +1,90 @@
+                    bits: 2048,
+                    policy: Policy {
+                        usage_flags: UsageFlags {
+                            sign_hash: true,
```
That should probs include `verify_hash` too?

hug-dev replied:
Yes will update all of that after

hug-dev commented:
Rebased and modified copyrights. The docs and content of the ECC/encrypt/decrypt/sign/verify is currently all copy-paste but this PR is only to create files/rename to make subsequent PRs easier.

ionut-arm reviewed (approved):
I'm not convinced having all the different encrypt operations under one name is ideal, but we can debate this further
